File: Basic_Math/power/power.rb
def power(num, pow)
    result = 1
    while (pow > 0)
        if (pow % 2 == 1)
            result = result * num
        end
        num = num * num
        pow = pow / 2
    end
    return result
end 

print("power:",power(2,3))

File: Basic_Math/power/power.js
const power  = (num, pow) => {
    let result = 1;
    while ( pow > 0 ) {
        if ( pow % 2 == 1 ) {
            result = result * num;
        }
        num = num * num;
        pow = parseInt(pow / 2);
    }
    return result;
}

console.log("power",power(2,3));